opentelemetry-api/src/opentelemetry/metrics/__init__.py

```diff
@@ -0,0 +1,245 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from re import ASCII, compile as compile_
+from typing import cast
+
+from opentelemetry.environment_variables import OTEL_PYTHON_METER_PROVIDER
+from opentelemetry.util.types import Attributes
+from abc import ABC, abstractmethod
+from logging import getLogger
+from opentelemetry.util._providers import _load_provider
+
+_logger = getLogger(__name__)
+
+
+class Measurement(ABC):
+
+    @abstractmethod
+    def __init__(self, value, **attributes: Attributes):
+        pass
+
+
+class Instrument(ABC):
+
+    _name_regex = compile_(r"[a-zA-Z][-.\w]{0,62}", ASCII)
+
+    @abstractmethod
+    def __init__(self, name, unit_of_measure="", description=""):
```

Comment:
minor: do you prefer `unit` or `unit_of_measure` as the name?

Reply:
I prefer `unit` :v: